### PR TITLE
User login

### PR DESCRIPTION
**Pull request by bramz**

### CODE REVIEW

File: static/templates/layout.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<!doctype html>
+<html>
+<head>
+  <meta charset="utf-8" />
+  <title>Gridpaste{% if paste %} - {{ paste.title }} {% endif %}</title>
+  <meta name="Description" content="A tool to paste functions, transformations, and structures on a geometric plane">
+  {% block head %}
+  <link rel="stylesheet" href="/css/build/main.min.css">
```
Comment (funkybob) [inline static/templates/layout.html]: Don't forget to use ```{% static %}``` to reference static assets.

File: user/models.py [hunk 2]
```diff
@@ -1,3 +1,7 @@
 from django.db import models
 
-# Create your models here.
+class User(models.Model):
+    id = models.CharField(max_length=255, primary_key=True)
```
Comment (funkybob) [inline user/models.py]: This seems a remarkably peculiar choice for a primary key.

What is generating these strings you'll be using for the values?

File: user/models.py [hunk 3]
```diff
@@ -1,3 +1,7 @@
 from django.db import models
 
-# Create your models here.
+class User(models.Model):
```
Comment (funkybob) [inline user/models.py]: Why have you avoided subclassing from one of the provided abstract user models, as described in the docs?

File: user/views.py [hunk 4]
```diff
@@ -1,26 +1,19 @@
...
 
 
 def register(request):
-    if request.method == 'GET':
-        template = get_template('register.html')
-        html = template.render()
-    
-        return HttpResponse(html)
\ No newline at end of file
+    if request.method == 'POST':
```
Comment (funkybob) [inline user/views.py]: It's so nice to see this long-form pattern for form processing is still remembered :)

File: user/models.py [hunk 5]
```diff
@@ -1,3 +1,7 @@
...
+    id = models.CharField(max_length=255, primary_key=True)
+    username = models.TextField()
+    password = models.TextField()
+    email = models.TextField()
```
Comment (funkybob) [inline user/models.py]: Does this site require having username distinct from email?

If you derive from `AbstractBaseUser` and declare `email` to be the `USERNAME_FIELD` then `UserCreationForm` will already include it.

File: user/models.py (hunk 2 above)
Comment (jahan-addison) [reply user/models.py]: In other words, why isnt this an integer type,

File: user/models.py (hunk 5 above)
Comment (jahan-addison) [reply user/models.py]: Yeah, i think ONLY usernameis fine. we dont need both anymore

File: user/models.py (hunk 2 above)
Comment (funkybob) [reply user/models.py]: Or omit an explicit key entirely, and get the AutoField django will give you by default.

**Comment by bramz** [comment]
So, taking all this into consideration and after doing some reading. I decided to go with django-registration as opposed to hand rolling registration. It adds to dependencies so if that is not the desired result, I can adjust accordingly.

File: static/templates/django_registration/registration_form.html [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{% extends 'layout.html' %}
+
+{% block content %}
+<table>
+    <form method='post' action=''>{% csrf_token %}
+        {{ form }}
+        <tr><td></td><td><input type="submit" value="Register" /></td>
```
Comment (jahan-addison) [inline static/templates/django_registration/registration_form.html]: what is this tr td shit?

Comment (bramz) [reply static/templates/django_registration/registration_form.html]: lol, sorry, i just grabbed this from some reference with all the html called by `django-registration`, i'll get rid of the table

**Pull request closed without merging**